add google parrot config

gompa opened the pull request:
enable google parrot in heads
add coreboot and linux config and patch to enable tpm0

tlaurion commented:
@gompa There was a long debate in the past under Heads, where it was decided that ifd could be distributed, where ME couldn't if not already part of the coreboot's blobs and or if the binary redistribution clauses were not clear on that effect.
This is why you can find extraction scripts in the blobs dir for boards requiring them.

gompa commented:
> @gompa There was a long debate in the past under Heads, where it was decided that ifd could be distributed, where ME couldn't if not already part of the coreboot's blobs and or if the binary redistribution clauses were not clear on that effect.
> This is why you can find extraction scripts in the blobs dir for boards requiring them.
> 
> The idea is that Heads git repository doesn't want to be the redistributor of such blobs.
> For the xx20 and x230 (maximized) those blobs are downloaded from Lenovo, cleaned, and extracted in the blobs dir.
> Purism scripts downloads them from their own hosting, having the redistribution liability.

but the ME and ifd are in the normale coreboot repo : https://github.com/coreboot/blobs/tree/master/mainboard/google/parrot,  this is just a nuterd version for more space, but i can remove them if needed

tlaurion commented:
@gompa 

```
user@x230-master:~/heads/build/coreboot-4.13/3rdparty/blobs/mainboard/google/parrot$ ls -al
total 2120
drwxr-xr-x 2 user user    4096 Nov 20 07:01 .
drwxr-xr-x 7 user user    4096 Nov 20 07:01 ..
-rw-r--r-- 1 user user    4096 Nov 20 07:01 descriptor.bin
-rw-r--r-- 1 user user 2093056 Nov 20 07:01 me.bin
-rw-r--r-- 1 user user   65536 Nov 20 07:01 snm_2130_coreboot.bin
```

Seems like those are redistributable blobs!

Tonux599 commented:
@gompa Please run the following in head's recovery shell and post the output to check everything is being measured okay.

```
source /etc/functions
pcrs
cbmem -1 | grep -B2 -i "pcr 2 measure"
cbfs -l -v 2>&1 | grep "File len\|File name"
```

gompa commented:
second command doesn't return anything so i guess not ? 

PCR-00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-01: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-02: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-03: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-04: 8A 6A 96 FD E1 A8 DD 96 27 14 79 DC 40 74 2B 36 AB A3 C2 B3 
PCR-05: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-06: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-07: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
File len               : 20
File name              : 'cbfs master header'
File len               : 122d4
File name              : 'fallback/romstage'
File len               : 6800
File name              : 'cpu_microcode_blob.bin'
File len               : 15607
File name              : 'fallback/ramstage'
File len               : 2f6
File name              : 'config'
File len               : 2b9
File name              : 'revision'
File len               : 2ff3
File name              : 'fallback/dsdt.aml'
File len               : 4a4
File name              : 'cmos_layout.bin'
File len               : 50b4
File name              : 'fallback/postcar'
File len               : 70d627
File name              : 'fallback/payload'
File len               : 1b818
File name              : ''
File len               : 10000
File name              : 'bootblock'

tlaurion commented:
> > @gompa There was a long debate in the past under Heads, where it was decided that ifd could be distributed, where ME couldn't if not already part of the coreboot's blobs and or if the binary redistribution clauses were not clear on that effect.
> > This is why you can find extraction scripts in the blobs dir for boards requiring them.
> > The idea is that Heads git repository doesn't want to be the redistributor of such blobs.
> > For the xx20 and x230 (maximized) those blobs are downloaded from Lenovo, cleaned, and extracted in the blobs dir.
> > Purism scripts downloads them from their own hosting, having the redistribution liability.
> 
> but the ME and ifd are in the normale coreboot repo : https://github.com/coreboot/blobs/tree/master/mainboard/google/parrot, this is just a nuterd version for more space, but i can remove them if needed

@gompa  I would suggest, sincerely and for simplicity, to simply copy paste the code of other blob extraction scripts and  implement download and verification logic on hashes obtained from a downloaded blob in repo for a specific commit, and reapply the logic of me_cleaner and verify the hashes of produced blobs dumped inside of your blobs dir, where coreboot config will look for the blobs at each build. Documenting those blobs inside of the board config would also be appreciated. And the build failing because of blobs not existing in tree would both respect the blobs non-redistribution clause, dodging legaility problems, while permitting other users to build easily.

The idea here is that if you implement this in this PR and the extraction tool can be called from command line, your board could then be added into CircleCI's builds, calling your script to download the blobs, neuter them and produce modified IFD matching the verified measurements you have taken to reproduce the same ROM. So each time a new commit lands into master, your board, as all other in CircleCI, will be rebuilt and ROMs produced. Endgoal of this is to have fwupd upgrades in the longer run, when CircleCI builds will have some logic added to be able to validate other builds from other instances and the hashes validated, prior of being uploaded into github releases, and github releases, ingested by LVFS so users can have upgrades of Heads upon new releases.

Tonux599 commented:
> second command doesn't return anything so i guess not ?
> 
> PCR-00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-01: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-02: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-03: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-04: 8A 6A 96 FD E1 A8 DD 96 27 14 79 DC 40 74 2B 36 AB A3 C2 B3
> PCR-05: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-06: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> PCR-07: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
> File len : 20
> File name : 'cbfs master header'
> File len : 122d4
> File name : 'fallback/romstage'
> File len : 6800
> File name : 'cpu_microcode_blob.bin'
> File len : 15607
> File name : 'fallback/ramstage'
> File len : 2f6
> File name : 'config'
> File len : 2b9
> File name : 'revision'
> File len : 2ff3
> File name : 'fallback/dsdt.aml'
> File len : 4a4
> File name : 'cmos_layout.bin'
> File len : 50b4
> File name : 'fallback/postcar'
> File len : 70d627
> File name : 'fallback/payload'
> File len : 1b818
> File name : ''
> File len : 10000
> File name : 'bootblock'

This will require further investigation. PCR2 should have measured Boot block, ROM stage, RAM stage, Heads linux kernel and initrd, so the fact its completely empty is quite concerning.

gompa commented:
after adding : CONFIG_TPM_MEASURED_BOOT=y
to the coreboot config:
```

TPM: Digest of FMAP: COREBOOT CBFS: bootblock to PCR 2 logged
TPM: Digest of FMAP: COREBOOT CBFS: fallback/romstage to PCR 2 logged
TPM: Digest of FMAP: COREBOOT CBFS: fallback/postcar to PCR 2 logged
TPM: Digest of FMAP: COREBOOT CBFS: fallback/ramstage to PCR 2 logged
TPM: Write digests cached in TCPA log to PCR
TPM: Write digest for FMAP: COREBOOT CBFS: bootblock into PCR 2
TPM: Write digest for FMAP: COREBOOT CBFS: fallback/romstage into PCR 2
TPM: Write digest for FMAP: COREBOOT CBFS: cmos_layout.bin into PCR 2
TPM: Write digest for FMAP: COREBOOT CBFS: fallback/postcar into PCR 2
TPM: Write digest for FMAP: COREBOOT CBFS: fallback/ramstage into PCR 2
TPM: Extending digest for FMAP: COREBOOT CBFS: cpu_microcode_blob.bin into PCR 2
TPM: Digest of FMAP: COREBOOT CBFS: cpu_microcode_blob.bin to PCR 2 measured
TPM: Extending digest for FMAP: COREBOOT CBFS: fallback/dsdt.aml into PCR 2
TPM: Digest of FMAP: COREBOOT CBFS: fallback/dsdt.aml to PCR 2 measured
TPM: Extending digest for FMAP: COREBOOT CBFS: cmos_layout.bin into PCR 2
TPM: Digest of FMAP: COREBOOT CBFS: cmos_layout.bin to PCR 2 measured
TPM: Extending digest for FMAP: COREBOOT CBFS: fallback/payload into PCR 2
TPM: Digest of FMAP: COREBOOT CBFS: fallback/payload to PCR 2 measured
 PCR-2 00f7afb8487a5f8bfaa6fbdb975cbdafec58e07c SHA1 [FMAP: COREBOOT CBFS: bootblock]
 PCR-2 a57f7e53ebfdf0e586aa1b33f3b94d0e30c672b3 SHA1 [FMAP: COREBOOT CBFS: fallback/romstage]
 PCR-2 d477da5c4846f6dad3ff374bb436c5292b2471e4 SHA1 [FMAP: COREBOOT CBFS: cmos_layout.bin]
 PCR-2 0f54280e678346f269d3f5bd35ca1ade1abe3493 SHA1 [FMAP: COREBOOT CBFS: fallback/postcar]
 PCR-2 d6ec8bec8d9e8684f646a5a0a64899ec86fda858 SHA1 [FMAP: COREBOOT CBFS: fallback/ramstage]
 PCR-2 8efffbbf487c06d4328ce4fbba2314554f2a2925 SHA1 [FMAP: COREBOOT CBFS: cpu_microcode_blob.bin]
 PCR-2 e065e93afa3904935395001928da02bac0d81efa SHA1 [FMAP: COREBOOT CBFS: fallback/dsdt.aml]
 PCR-2 d477da5c4846f6dad3ff374bb436c5292b2471e4 SHA1 [FMAP: COREBOOT CBFS: cmos_layout.bin]
 PCR-2 9e3ab61673c876f1aa35f6c202c6a53ff2090a42 SHA1 [FMAP: COREBOOT CBFS: fallback/payload]

```

Tonux599 commented:
@gompa sorry can you just confirm that `pcrs` show PCR-2 being populated?

gompa commented:
yes ! : 
```
PCR-00: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-01: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-02: 3D C8 9A F6 49 50 72 A4 91 06 1F 54 F0 72 C3 B6 0F 42 FF 79 
PCR-03: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-04: 8A 6A 96 FD E1 A8 DD 96 27 14 79 DC 40 74 2B 36 AB A3 C2 B3 
PCR-05: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-06: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
PCR-07: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
```

tlaurion commented on config/coreboot-parrot.config:
```diff
@@ -0,0 +1,20 @@
+CONFIG_PAYLOAD_FILE="../../build/Parrot/bzImage"
+CONFIG_LINUX_COMMAND_LINE="intel_iommu=igfx_off quiet"
+CONFIG_LINUX_INITRD="../../build/Parrot/initrd.cpio.xz"
+CONFIG_USE_OPTION_TABLE=y
+CONFIG_STATIC_OPTION_TABLE=y
```
Wait.... That works?
That was a theoritical blocker for https://github.com/osresearch/heads/pull/944

So this boots? Not sure its required here unless the board requires users to modify  the defaults locally, as required per some users under t430

tlaurion replied:
So remove those please.

gompa replied:
done

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,55 @@
+# Configuration for a x230 running Qubes and other OSes
+# STATIC_OPTION_TABLE is set inside of coreboot config
```
Not sure required. IF not, please modify

gompa replied:
sorry, modify to what ?

tlaurion replied:
@gompa if related optionsnot required in coreboot, remove there and here.

tlaurion commented on config/linux-parrot_common.config:
```diff
@@ -0,0 +1,304 @@
+CONFIG_LOCALVERSION="-heads"
+# CONFIG_LOCALVERSION_AUTO is not set
+CONFIG_KERNEL_XZ=y
+# CONFIG_SWAP is not set
+# CONFIG_CROSS_MEMORY_ATTACH is not set
+CONFIG_NO_HZ_IDLE=y
+CONFIG_PREEMPT_VOLUNTARY=y
+CONFIG_LOG_BUF_SHIFT=18
+CONFIG_BLK_DEV_INITRD=y
+CONFIG_INITRAMFS_SOURCE="../../../blobs/dev.cpio"
+# CONFIG_RD_GZIP is not set
+# CONFIG_RD_BZIP2 is not set
+# CONFIG_RD_LZMA is not set
+# CONFIG_RD_LZO is not set
```
All that file was created with the output of make savedefconfig in linux build dir or not?
If not please to and overwrite

gompa replied:
i think this is a copy from linux-librem_common.config with +CONFIG_TCG_TIS_I2C_NUVOTON=y
but i dont think that option is needed, this machine needs other modules for networking ill create a new kernel config for it tomorrow

gompa replied:
new version is created with savedefconfig, and has eth0 working

gompa commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,55 @@
+export CONFIG_BOOTSCRIPT=/bin/gui-init
+export CONFIG_BOOT_REQ_HASH=n
+export CONFIG_BOOT_REQ_ROLLBACK=n
+export CONFIG_BOOT_KERNEL_ADD="intel_iommu=on intel_iommu=igfx_off"
```
not sure, was a direct copy from a other config file, will test without

gompa replied:
works fine without it

gompa commented:
@tlaurion 
> @gompa I would suggest, sincerely and for simplicity, to simply copy paste the code of other blob extraction scripts and implement download and verification logic on hashes obtained from a downloaded blob in repo for a specific commit, and reapply the logic of me_cleaner and verify the hashes of produced blobs dumped inside of your blobs dir, where coreboot config will look for the blobs at each build. Documenting those blobs inside of the board config would also be appreciated. And the build failing because of blobs not existing in tree would both respect the blobs non-redistribution clause, dodging legaility problems, while permitting other users to build easily.
> 
> The idea here is that if you implement this in this PR and the extraction tool can be called from command line, your board could then be added into CircleCI's builds, calling your script to download the blobs, neuter them and produce modified IFD matching the verified measurements you have taken to reproduce the same ROM. So each time a new commit lands into master, your board, as all other in CircleCI, will be rebuilt and ROMs produced. Endgoal of this is to have fwupd upgrades in the longer run, when CircleCI builds will have some logic added to be able to validate other builds from other instances and the hashes validated, prior of being uploaded into github releases, and github releases, ingested by LVFS so users can have upgrades of Heads upon new releases.

this should be done now, please let me know if anything else needs changing

tlaurion commented on blobs/parrot/download_clean_me.sh:
```diff
@@ -0,0 +1,49 @@
+extractdir=$(mktemp -d)
+cd "$extractdir"
+
+echo "### Downloading: https://github.com/coreboot/blobs/raw/master/mainboard/google/parrot/me.bin..."
```
@gompa Stated like this, it leaves the door open for the downloaded binary to not match what is in repo in the future and will require maintainership.

Its a choice!

tlaurion replied:
Also, in case the binary changes, the shinked ifd might not fit later on

gompa replied:
fixed

tlaurion commented on blobs/parrot/download_clean_me.sh:
```diff
@@ -0,0 +1,49 @@
+
+FINAL_ME_BIN_SHA256SUM="e985feb4a2879a99fb792f2d425c17a68ee07ba8bc0fd39a7f9eb65d8c6d5f11 $BLOBDIR/me.bin"
+ME_SHA256SUM="1790fabc16afc36ab1bcfd52e10b805855d5e2a4eb96ea78781ffb60a0941928  me.bin"
+ME8_5M_PRODUCTION_SHA256SUM="821c6fa16e62e15bc902ce2e958ffb61f63349a471685bed0dc78ce721a01bfa  app/ME8_5M_Production.bin"
```
Irrelevant. Please delete

gompa replied:
i fixed this in a later commit

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,53 @@
+# Configuration for a Parrot running Qubes and other OSes
+#
```
A note here on how you obtained and reproduced the shrinked ifd would be helpful for people wanting to reproduce.

tlaurion replied:
@gompa would still be required!

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,53 @@
+#SSH client/server
+CONFIG_DROPBEAR=n
+#Ethernet driver (Heads only)
+CONFIG_LINUX_E1000E=n
```
Have checked the specs but no GBE in rom means that creating a full rom like we do here with ME and ifd results in the equivalent of a maximized board.

Putting ME and creating an expended ifd taking advantage of the freed space means that the IFD put into repo is unlocked with ifdtool. is that the case?

If such, the flshrom upgrade line into board config should remove `--ifd --image BIOS` as in maximized boards.

Question is: you were successful flashing it externally with ifd and ME included in coreboot image? If there is no Ethernet port, and then no GBE region needed here, then it is candidate for maximized board and the flash options below should be changed.

gompa replied:
> Question is: you were successful flashing it externally with ifd and ME included in coreboot image? If there is no Ethernet port (I read eth0 somethwere... No need for a gbe blob?), and then no GBE region needed here, then it is candidate for maximized board and the flash options below should be changed accordingly. If there is eth0, its not e1000e based?

yeah i was succesfull flashing it externaly, first time i flashed it internaly  i "bricked" it. (probably locked ifd)
the nic is a tg3 broadcom card and i dont think it needs a gbe blob

> 
> Otherwise, if IFD was not unlocked (for some reason) and we do not want to be able to upgrade ME later on or reflash whole SPI region internally, leave it be with specifying --ifd --image BIOS below. This means that on firmware future internal firmware upgrades, only the BIOS region (fixed in both ifd provided matching CBFS_REGION in coreboot config) will be modifiable. If ME is upgraded in repo later on, and IFD was not unlocked, it means that external reflashing would be required to write ifd, ME changing size.

the ifd is not unlocked mostly because i forgot to unlock it and flashed it externaly , i could unlock it ? i dont know what's prefered

tlaurion replied:
@gompa: Other boards (non maximized) take into consideration that blobs are extracted from backups (could be another script called extract, as in x230 maximized blobs dir), where the script calls ifdtool -u by default, where the user can decide for himself to not unlock it.

If the goal is to have maintained fwupd board, where a new commit of Heads could lead to new ME bin in coreboot in newer coreboot version, then only unlocked ifd with corrected new maximized free space + updated ME (from unlocked region) would permit such upgrade. (Code is pending review under QubesOS and FWUPD as of right now, where helper script will need to be developped under Heads to verify and ask user to flash on later version.)

Not unlocking ifd means that the board config would need to stay as is only specifying BIOS to be flashable, where future ME upgrade (if that is even a thing, I have not researched this board at all) would require the board to be externally reflashed. Is there a screw that needs to be unscrewed to flash SPI? I have no idea here, so I leave you the choice of implementation.

In all case, if we want a fully upgradeable ROM from fwupd, it would need ifd to be unlocked. 
There is no consensus on this as of now and there was debates on the subject before around, if you look. My opinion is that IFD should be unlocked so whole SPI can be inspected/upgraded.

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,53 @@
+export CONFIG_BOOT_KERNEL_REMOVE="quiet"
+export CONFIG_BOOT_DEV="/dev/sda2"
+export CONFIG_BOARD_NAME="Parrot"
+export CONFIG_FLASHROM_OPTIONS="--force --noverify-all -p internal --ifd --image bios"
```
Change to "--force --noverify-all -p internal" if BIOS image is omplete, no GBE is needed.
If GBE image needed, another extraction script needs to be put in blobs directory, so that the user has to run it to extract his GBE from original rom backup, and coreboot config needs to depend on it.

Otherwise, we have another maximized board, where CBFS region should be reviewed in coreboot config, and ifd reviewed one last time to make sure that BIOS region is taking all the space left from ME.

tlaurion replied:
On X230, the absence of GBE makes the initialization of the ethernet card impossible, since the MAC address is defined under GBE blob. That is why I ask so many questions about it... And to document this board's PR so others can review steps needed to produce a functional port inside of Heads.

Thanks for your contribution!

tlaurion commented:
@gompa another review round in comments!

tlaurion commented:
One nice to have addition for board not needing external blobs or having redistribution of blobs issue (like here) is to have a CircleCI entree added into .circleci/config.yml:

```
      - run:
          name: Download and neuter xx30 ME (keep generated GBE and extracted IFD in tree)
          command: |
            ./blobs/xx30/download_clean_me.sh
      - run:
          name: x230-maximized
          command: |
            rm -rf build/x230-maximized/* build/log/* && make CPUS=4 V=1 BOARD=x230-maximized || touch /tmp/failed_build
          no_output_timeout: 3h
      - run:
          name: Output build failing logs
          command: |
            if [[ -f /tmp/failed_build ]]; then find ./build/ -name "*.log" -type f -mmin -1|while read log; do echo ""; echo '==>' "$log" '<=='; echo ""; cat $log;done; exit 1;else echo "Not failing. Continuing..."; fi
      - run:
          name: Output x230-maximized hashes
          command: |
            cat build/x230-maximized/hashes.txt \
      - run:
          name: Archiving build logs for x230-maximized
          command: |
            tar zcvf build/x230-maximized/logs.tar.gz ./build/log/*
      - store-artifacts:
          path: build/x230-maximized
```

Then if you are following your github project from CircleCI, the build will happen on next commit.
And if that next commit pushed includes the above section calling your blob script prior of building your board, CircleCI will contain produced roms, with parrot included.

gompa commented:
@tlaurion ah cool wil have to check that out ! sure you can add me to #692.
the board is a acer c710 (chromebook Parrot) a basic sandy bridge laptop(there could be a ivy bridge version not a 100% sure about that)
it lacks most cpu extentions needed for qubes but this is what i have available, i'd like to try some more recent chromebooks to get heads running on some "cheap" recent hardware with decent batterylife, and i think the chrome book ec is opensource too so that would be a plus

tlaurion commented:
> @tlaurion ah cool wil have to check that out ! sure you can add me to #692.
> the board is a acer c710 (chromebook Parrot) a basic sandy bridge laptop(there could be a ivy bridge version not a 100% sure about that)
> it lacks most cpu extentions needed for qubes but this is what i have available, i'd like to try some more recent chromebooks to get heads running on some "cheap" recent hardware with decent batterylife, and i think the chrome book ec is opensource too so that would be a plus

@gompa running qubes-hcl-report would resolve the doubts with clear viww out of dom0 console over qubesos. Otherwise, its just speculations.

gompa commented:
> > @tlaurion ah cool wil have to check that out ! sure you can add me to #692.
> > the board is a acer c710 (chromebook Parrot) a basic sandy bridge laptop(there could be a ivy bridge version not a 100% sure about that)
> > it lacks most cpu extensions needed for qubes but this is what i have available, i'd like to try some more recent chromebooks to get heads running on some "cheap" recent hardware with decent batterylife, and i think the chrome book ec is opensource too so that would be a plus
> 
> @gompa running qubes-hcl-report would resolve the doubts with clear viww out of dom0 console over qubesos. Otherwise, its just speculations.

when booting the qubes installer from usb it gives an error about unsupported hardware, missing features IOMMU/VT-d/AMD-Vi,Interupt Remapping 
missing vt-d is confirmed by the intel arkpage: https://ark.intel.com/content/www/us/en/ark/products/56056/intel-celeron-processor-847-2m-cache-1-10-ghz.html
support for this chromebook has ended, most info i found on other still supported chromebooks stated they have intel vt-d enabled

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,53 @@
+# Configuration for a Parrot running Qubes and other OSes
```
- No QubesOS support since no vt-d support!

tlaurion commented on boards/Parrot/Parrot.config:
```diff
@@ -0,0 +1,53 @@
+# Configuration for a Parrot running Other OSes, No QubesOS support since no vt-d support!
+#
+# Deactivated to fit in coreboot's CONFIG_CBFS_SIZE=0x700000 :
```
doestn match coreboot config line

gompa commented:
@tlaurion i cleaned up the board config, could you please check if the ci config is correct like this, its a direct copy of the snippet above but i cant test

tlaurion commented:
@gompa Sorry. Got lost in time.

Created branch https://github.com/tlaurion/heads/tree/gompa_parrot which should trigger the CircleCI build https://app.circleci.com/pipelines/github/tlaurion/heads/717/workflows/d2423e2b-21a3-4a7f-b86c-fb0a55981d85/jobs/864

But CircleCI builds are foo bar for the moment. Too many boards and parallelization is needed at this point.
Blockers #977 #935 to gain precious compilation time.

tlaurion commented:
@gompa meanwhile, Heads community in not really loving PRs with many commits (I do. I think this board is the exact example of how to add a new boards into Heads).

@MrChromebox : Would you be ok with merging so many commits directly? 
@gompa I could squash, but that removes the credit to you and the steps taken to make it work, while this PR would show the individual commits.

MrChromebox commented:
@tlaurion absolutely not, this definitely needs cleanup.

Additionally, I'm not certain this will work for all Parrot boards given there are both SNB/IVB versions, which I believe use different ME firmware. I can get away with it for my MrChromebox UEFI firmware since I don't reflash the ME

MrChromebox commented on config/coreboot-parrot.config:
```diff
@@ -0,0 +1,18 @@
+CONFIG_TPM_RDRESP_NEED_DELAY=y
+CONFIG_UART_PCI_ADDR=0
+CONFIG_NO_GFX_INIT=y
+CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x80000
```
these aren't needed

MrChromebox commented on config/linux-parrot_common.config:
```diff
@@ -0,0 +1,302 @@
+CONFIG_LOCALVERSION="-heads"
```
why `parrot-common`? I'd drop `-common`

MrChromebox commented on patches/coreboot-4.13/0002-enable-tpm-parrot.patch:
```diff
@@ -0,0 +1,14 @@
+										device pnp ff.1 on # dummy address
+										end
+								end
++ 								chip drivers/pc80/tpm
```
this should be upstreamed to coreboot

tlaurion commented:
@gompa ping!

tlaurion commented:
@gompa ping
